pkg/router/base_rule.go

```diff
@@ -206,10 +206,21 @@ func (rri *RouteRuleImplBase) RedirectRule() api.RedirectRule {
 // Select Cluster for Routing
 // if weighted cluster is nil, return clusterName directly, else
 // select cluster from weighted-clusters
-func (rri *RouteRuleImplBase) ClusterName() string {
+func (rri *RouteRuleImplBase) ClusterName(ctx context.Context) string {
 	if len(rri.weightedClusters) == 0 {
-		return rri.defaultCluster.clusterName
+		// If both 'cluster_name' and 'cluster_variable' are configured, 'cluster_name' is preferred.
+		if rri.defaultCluster.clusterName != "" {
+			return rri.defaultCluster.clusterName
+		} else if clusterName, err := variable.GetVariableValue(ctx, rri.routerAction.ClusterVariable); err == nil {
```

Comment:
else if -> if

Reply:
if err != nil , should return "error string" or "" ?

Reply:
> if err != nil , should return "error string" or "" ?

If an error occurred while get cluster name from variable, then we need to get it from `DefaultCluster`.

Reply:
> else if -> if

done.